### DISCUSSION TITLE
Best practice for refactoring using TypeScript with SFC?

### DISCUSSION THREAD

**Discussion by baixiaoyu2997** [nuxt/nuxt#25415] [Questions] [answered]
When I use vscode's refactoring feature, it behaves inconsistently in vue files and ts files
![image](https://github.com/nuxt/nuxt/assets/9149999/62cf1d65-d68f-447d-833d-ef5e9c67c8a3)
I can refactor `test.bb` to `test.cc` in vue file (ts file also changed)
![image](https://github.com/nuxt/nuxt/assets/9149999/15de1b6a-dc2c-4b5a-b57a-84a87cc898a9)
but I can't do it the other way around (vue file has not changed)
![image](https://github.com/nuxt/nuxt/assets/9149999/b28fc40d-f110-4554-bbc1-2bd620868c40)

I have volar installed and enabled
What should I do?

**Comment by productdevbook** [comment]
```
pnpm nuxt prepare
or 
yarn nuxt prepare 
...
```

**Reply by baixiaoyu2997** [reply]
I tested it while running the dev server

**Comment by baixiaoyu2997** [comment, marked as answer]
I know the problem. If I want to achieve the effect I want, I need to enable takeover mode.
